core/Network/TLS/Types.hs

```diff
@@ -19,6 +19,8 @@ module Network.TLS.Types
 import Data.ByteString (ByteString)
 import Data.Word
 
+type HostName = String
```

Comment:
Then in the PR I'd rather keep this 5th local definition of a type alias for now and remove them all when a remplacement is ready.

Does the `Hostname` plan include something for IDN support?